Source/WebCore/editing/Editor.cpp

```diff
@@ -2105,8 +2105,54 @@ void Editor::selectComposition()
     document().selection().setSelection(selection, { });
 }
 
+Element* Editor::selectedElement()
+{
+    Ref document = protectedDocument();
+
+    RefPtr node = document->selection().selection().end().protectedDeprecatedNode();
+    auto* result = dynamicDowncast<Element>(node.get());
+    if (result)
+        return result;
+
+    return node->parentElement();
+}
+
+void Editor::removeWritingSuggestionIfNeeded()
+{
+    Ref document = protectedDocument();
+    document->updateStyleIfNeeded();
+
+    m_customCompositionAnnotations = { };
+    m_isHandlingAcceptedCandidate = false;
+
+    RefPtr selectedElement = this->selectedElement();
+    if (!selectedElement)
+        return;
+
+    selectedElement->setWritingSuggestion(nullString());
+    selectedElement->invalidateStyleAndRenderersForSubtree();
+
+    // These following lines are needed for AX compatibility:
+
+    RefPtr node = document->selection().selection().end().protectedDeprecatedNode();
+    if (!node)
+        return;
+
+    RefPtr text = dynamicDowncast<Text>(node.get());
+    if (!text)
+        return;
+
+    if (CheckedPtr cache = document->existingAXObjectCache())
+        cache->onTextCompositionChange(*selectedElement, CompositionState::Ended, true, text->data(), m_compositionStart, m_isHandlingAcceptedCandidate);
```

Comment:
This also seems a little strange, but I'm not particularly familiar with AX machinery. I expect it would be preferred to introduce a new, more-specific method to achieve your desired behavior.